test-suite/handwritten-src/cpp/test_helpers.cpp

```diff
@@ -161,4 +168,42 @@ std::vector<uint8_t> TestHelpers::id_binary(const std::vector<uint8_t> & v) {
     return v;
 }
 
+snapchat::djinni::Future<int32_t> TestHelpers::get_async_result() {
+    auto* p = new snapchat::djinni::Promise<int32_t>();
+    auto f = p->getFuture();
+
+#if defined(__EMSCRIPTEN__)
+    emscripten_async_call([] (void* context) {
+        auto* p = reinterpret_cast<snapchat::djinni::Promise<int32_t>*>(context);
+        p->setValue(42);
+        delete p;
+    }, p, 10/*ms*/);
+#else
+    std::thread t([p] () mutable {
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+        p->setValue(42);
+        delete p;
+    });
+    t.detach();
+#endif
+    return f;
+}
+
+snapchat::djinni::Future<std::string> TestHelpers::future_roundtrip(snapchat::djinni::Future<int32_t> f) {
+    return f.then([] (snapchat::djinni::Future<int32_t> f) {
+        return std::to_string(f.get());
+    });
+}
+
+snapchat::djinni::Future<std::string> TestHelpers::check_async_interface(const std::shared_ptr<AsyncInterface> & i) {
```

Comment:
Are there any C++ unit tests calling this?

I don't see any pure C++ unit tests.  I also don't see an tests on calls to setException, but might have just missed this one.

Reply:
there is a `testFutureRoundtripWithException` test that throws an exception in platform code which will indirectly `setException`